docs/tutorials/programmatic-upgradeable-module.md

```diff
@@ -0,0 +1,266 @@
+---
+title: "Programmatic Upgradeable Module"
+slug: "programmatic-upgradeable-module"
+---
+
+# Programmatic Upgradeable Module
+
+This tutorial will go over two sections:
+1. How to publish modules to a resource account.
+2. How to upgrade modules in a resource account.
+
+A [resource account](https://aptos.dev/move/move-on-aptos/resource-accounts/) is a developer feature used to manage resources independent of an account managed by a user,
+specifically publishing modules and providing on-chain-only access control, e.g. signers.
+
+Typically, a resource account is used for two main purposes:
+
+- Store and isolate resources; a module creates a resource account just to host specific resources.
+- Publish module as a standalone (resource) account, a building block in a decentralized design where no private keys can control the resource account. The ownership (SignerCap) can be kept in another module, such as governance.
+
+The first step will go over section 1: **How to publish modules to a resource account.**
+
+## 1. How to publish modules to a resource account
+
+Before publishing the module, we want to create a resource account prior, so that the module then gets uploaded to the resource account. To do this, we run the command:
+
+1. `aptos move create-resource-account-and-publish-package --address-name <your-address> --seed <seed>`
+    1. Replace `<your-address>` with your account address.
+    2. Replace `<seed>` with a seed that only you know. This is used as a custom input to generating the resource account address.
+
+When running this command, this will **FIRST calculate** your resource account address. 
+A prompt below will show:
+
+```jsx
+Do you want to publish this package under the resource account's address 
+0x090ad1536fe5cfcb5632b3026f99f8415c55b69ce54b6f17ed8cd7edbcb5edfa? [yes/no]
+```
```

Comment:
This seems like a poor experience.  Run a command, and kill it in the middle, then do something else.

You should be able to take advantage of the following command to get the address ahead of time.
```
aptos account derive-resource-account-address --seed <seed>
```

Reply:
Thanks for this neat command, added this in with instructions!